Support SO_RXQ_OVFL socket option (android/fuchsia/linux)

junhochoi opened the pull request:
This PR implements support of RXQ_OVFL flag and parsing ControlMessage
to get the packet drop counter of UDP socket.

asomers commented on src/sys/socket/mod.rs:
```diff
@@ -826,6 +842,14 @@ pub enum ControlMessage<'a> {
               target_os = "ios",))]
     Ipv6PacketInfo(&'a libc::in6_pktinfo),
+
+    // SO_RXQ_OVFL indicates that an unsigned 32 bit value
```
Did you mean to make this a doc comment instead of a regular comment?

junhochoi replied:
Changed to `///`.

asomers commented on src/sys/socket/mod.rs:
```diff
@@ -988,6 +1020,8 @@ impl<'a> ControlMessage<'a> {
               target_os = "android", target_os = "ios",))]
             ControlMessage::Ipv6PacketInfo(_) => libc::IPPROTO_IPV6,
+            #[cfg(target_os = "linux")]
+            ControlMessage::RxqOvfl(_) => libc::SO_RXQ_OVFL,
```
Could you double-check this part?

junhochoi replied:
I updated the condition to include `android`, `fuchsia`, `linux`. CI seems running ok.

asomers replied:
I mean that `SO_RXQ_OVFL` doesn't look like a valid value for `cmsg_level`.  Is that really correct?

junhochoi replied:
Oh right. updated to `libc::SOL_SOCKET`. thanks!

asomers reviewed (changes requested):
Don't forget to add a CHANGELOG entry.  Also, what about Android and Fuchsia?  libc defines this constant for them.

junhochoi commented:
> Don't forget to add a CHANGELOG entry. Also, what about Android and Fuchsia? libc defines this constant for them.

I added CHANGELOG entry and addressed other comments. Thanks!

asomers commented on CHANGELOG.md:
```diff
@@ -12,6 +12,9 @@ This project adheres to [Semantic Versioning](https://semver.org/).
   (#[1456](https://github.com/nix-rust/nix/pull/1456))
 - Added `TcpUserTimeout` socket option (sockopt) on Linux and Fuchsia.
   (#[1457](https://github.com/nix-rust/nix/pull/1457))
+  (#[1447](https://github.com/nix-rust/nix/pull/1447))
```
Somehow you duped a line here.
```suggestion
```

junhochoi replied:
Fixed (I don't know where it comes from! probably something during rebase with master) Thanks

asomers reviewed (approved):
bors r+

asomers commented:
The tests are failing due to a just-landed PR on head.

junhochoi commented:
> The tests are failing due to a just-landed PR on head.

rebased to master and pushed a fix.

junhochoi commented:
@asomers the build is green. should be good to go.

asomers reviewed (approved):
bors r+

asomers commented:
bors retry

junhochoi commented:
rebased to master again.

asomers reviewed (approved):
bors r+
